### ISSUE TITLE
manual midpoint should also trigger on (a+b)*0.5

### ISSUE THREAD

**Issue by oscargus** [rust-lang/rust-clippy#17022]
### What it does

It is easy to write this thinking that it will avoid division. At least with higher optimization levels, (a + b)/2.0 will be transformed to (a + b)*0.5, so it is basically the same expression.

(One can even consider writing midpoint using `*0.5`.)

### Advantage

Overflow etc.

### Drawbacks

Same as midpoint.

### Example

```rust
let c = (a + b)*0.5;
```

Could be written as:

```rust
let c = a.midpoint(b);
```

### Comparison with existing lints

Same as `manual_midpoint`.

### Additional Context

https://github.com/rust-lang/rust/pull/156647

**Comment by oscargus** [comment]
It should probably trigger on `(a + b) >> 1` for unsigned integer types as well, which gives the same compiler output as `(a + b) / 2`. I'll see if I can update #17025 to cover that as well. See https://godbolt.org/z/eav1PjPhx (same for u32, not same for i32).
